exporters/provider_common/openshift.py

```diff
@@ -1,11 +1,61 @@
+import logging
+import re
+import time
 from datetime import datetime
-from typing import Union
+from typing import Dict, List, Optional, Set, Tuple, Union
+
+from openshift.dynamic import DynamicClient, ResourceInstance
+from openshift.dynamic.exceptions import ResourceNotFoundError
+from openshift.dynamic.resource import ResourceField
 
 from pelorus.timeutil import parse_assuming_utc
 
 # https://docs.openshift.com/container-platform/4.10/rest_api/objects/index.html#io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta
 _DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%SZ"
 
+SUPPORTED_REPLICA_OBJECTS = ["ReplicaSet", "ReplicationController"]
+
+# Cache threshold for each of the cached_parents_dict entries
+CACHE_THRESHOLD = 60 * 60 * 24
+cached_parents_dict: dict[str, ResourceInstance] = {}
```

Comment:
do not like having two things, let's merge them together 

we can use tuple, dict or a pydantic object as the dict value

example

```py
cached_parents = {
    '123': (<k8_obj_reference>, 123456)
}
```

Reply:
merged.

Reply:
the typing is inverted, its a tuple of first item being openshift object and the second being a float representing a timestamp, right?

Reply:
good catch !